internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -964,6 +965,15 @@ func expandArmServerNetwork(d *pluginsdk.ResourceData) *servers.Network {
 		network.PrivateDnsZoneArmResourceId = utils.String(v.(string))
 	}
 
+	// `d.GetOk()` cannot identify if the bool property `public_network_access_enabled` is set or not in the tf config since d.GetOk() always returns `false` when `public_network_access_enabled` is set to `false` and `public_network_access_enabled` isn't set in the tf config
+	if !pluginsdk.IsExplicitlyNullInConfig(d, "public_network_access_enabled") {
+		publicNetworkAccessEnabled := servers.ServerPublicNetworkAccessStateDisabled
+		if d.Get("public_network_access_enabled").(bool) {
+			publicNetworkAccessEnabled = servers.ServerPublicNetworkAccessStateEnabled
+		}
+		network.PublicNetworkAccess = pointer.To(publicNetworkAccessEnabled)
+	}
```

Comment:
By defaulting this in the Read we can just pull from the config:

```suggestion
	// `d.GetOk()` cannot identify if the bool property `public_network_access_enabled` is set or not in the tf config since d.GetOk() always returns `false` when `public_network_access_enabled` is set to `false` and `public_network_access_enabled` isn't set in the tf config
	publicNetworkAccessEnabled := servers.ServerPublicNetworkAccessStateEnabled 
	if !d.Get("public_network_access_enabled").(bool) {
		publicNetworkAccessEnabled = servers.ServerPublicNetworkAccessStateDisabled
	}
	network.PublicNetworkAccess = pointer.To(publicNetworkAccessEnabled)
```

**Note**: since the Azure API defaults this to `true` today, we'll also need to update the Read function to account for this too.

Reply:
As I above mentioned, we can't simply hard-code the default value for public_network_access_enabled. So I assume that we can't apply your this suggestion.

Reply:
Yes, we can.

Reply:
Updated